infra/blockchain/evmErrors.go

```diff
@@ -0,0 +1,193 @@
+// "transaction gas price {} less than the minimum value {}"
+// "Tx with same nonce already inserted. To replace it, you need to specify a gas price > {}"
+// ready_account_pool and deferred_pool are inconsistent! ready_tx={:?} first_pending={:?}
+// "Failed to read account_cache from storage: {}"
+// "Transaction {:?} is discarded due to in too distant future",
+// "Transaction {:?} is discarded due to a too stale nonce",
+// "Transaction {:?} is discarded due to out of balance, needs {:?} but account balance is {:?}",
+// "Failed imported to deferred pool
+// "tx already exist"
+
+func GetRpcErrorType(errorStr string) TxRpcError {
+	if strings.Contains(errorStr, "txpool is full") {
+		return TX_ERR_RPC_TXPOOL_FULL
+	}
+	if strings.Contains(errorStr, "Transaction Pool is full") {
+		return TX_ERR_RPC_TXPOOL_FULL
+	}
+	if strings.Contains(errorStr, "failed to insert tx into pool (validation failed)") {
+		return TX_ERR_RPC_INSERT_VALIDATION_FAIL
+	}
+	if strings.Contains(errorStr, "failed to insert tx into pool (quota not enough)") {
+		return TX_ERR_RPC_INSERT_FAIL_QUOTA_UNENOUGH
+	}
+	if strings.Contains(errorStr, "exceeds the maximum value") {
+		return TX_ERR_RPC_EXCEED_MAX_GAS
+	}
+	if strings.Contains(errorStr, "Tx with same nonce already inserted") {
+		return TX_ERR_RPC_SAME_NONCE_ALREADY_INSERTED
+	}
+	if strings.Contains(errorStr, "ready_account_pool and deferred_pool are inconsistent!") {
+		return TX_ERR_RPC_POOL_INCONSISTENT
+	}
+	if strings.Contains(errorStr, "Failed to read account_cache from storage") {
+		return TX_ERR_RPC_FAIL_READ_ACCOUNT_CACHE
+	}
+	if strings.Contains(errorStr, "discarded due to in too distant future") {
+		return TX_ERR_RPC_NONCE_DISTANT
+	}
+	if strings.Contains(errorStr, "discarded due to a too stale nonce") {
+		return TX_ERR_RPC_STALE_NONCE
+	}
+	if strings.Contains(errorStr, "discarded due to out of balance") {
+		return TX_ERR_RPC_OUT_OF_BALANCE
+	}
+	if strings.Contains(errorStr, "NotEnoughCash") {
+		return TX_ERR_RPC_ESTIMATE_NOT_ENOUGH_CASH
+	}
+	if strings.Contains(errorStr, "block_number is missing for best_hash") {
+		return TX_ERR_RPC_MISSING_BEST_HASH
+	}
+	if strings.Contains(errorStr, "Failed imported to deferred pool") {
+		return TX_ERR_RPC_FAILED_IMPORT_TO_DEFFER_POOL
+	}
+	if strings.Contains(errorStr, "tx already exist") {
+		return TX_ERR_RPC_TX_ALREADY_EXIST
+	}
+	if strings.Contains(errorStr, "Vm reverted") || strings.Contains(errorStr, "VmError") || strings.Contains(errorStr, "Transaction reverted") {
+		return TX_ERR_RPC_ESTIMATE_VM_ERR
+	}
+	if strings.Contains(errorStr, "too many requests") {
+		return TX_ERR_RPC_TOO_MANY_REQUEST
+	}
+	if strings.Contains(errorStr, "allowed qps exceeded") {
+		return TX_ERR_RPC_CONFURA_QPS_EXCEED
+	}
+	if strings.Contains(errorStr, "daily request count exceeded") {
+		return TX_ERR_RPC_CONFURA_DAILYQ_EXCEED
+	}
+	if strings.Contains(errorStr, "EpochHeightOutOfBound") {
+		return TX_ERR_RPC_EPOCH_HEIGHT_OUT_BOUND
+	}
+
+	return TX_ERR_RPC_OTHER
```

Comment:
seems we can also use a hashmap as configuration table or we can use switch statement to be more concise.